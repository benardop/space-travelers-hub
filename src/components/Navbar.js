import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assets/planet.png';
import './stylesheets/Navbar.css';

const navLinkStyles = ({ isActive }) => ({
  fontWeight: isActive ? 'bold' : 'normal',
  textDecoration: isActive ? 'underline' : 'none',
});

const Header = () => (
  <header>
    <div className="logo">
      <img className="space-logo" src={logo} alt="logo" />
      <span className="title-logo">Space Travelers&apos;  Hub</span>
    </div>
    <nav>
      <ul>
        <NavLink to="/" style={navLinkStyles}><span className="rockets">Rockets</span></NavLink>
        <NavLink to="missions" style={navLinkStyles}><span className="missions">Missions</span></NavLink>
        <NavLink to="myprofile" style={navLinkStyles}><span className="myprofile">My Profile</span></NavLink>
      </ul>
    </nav>
  </header>
);

export default Header;
