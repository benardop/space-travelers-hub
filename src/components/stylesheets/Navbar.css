@import '../../App.css';

header {
  height: 80px;
  margin: 0 auto;
  display: flex;
  padding: 1rem 0;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  box-shadow: rgba(0, 0, 0, 0.07) 0 1px 2px, rgba(0, 0, 0, 0.07) 0 2px 4px, rgba(0, 0, 0, 0.07) 0 4px 8px, rgba(0, 0, 0, 0.07) 0 8px 16px, rgba(0, 0, 0, 0.07) 0 16px 32px, rgba(0, 0, 0, 0.07) 0 32px 64px;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
  background-color: var(--color-white);
}

.logo {
  display: flex;
  align-items: center;
  cursor: pointer;
  margin-left: 2rem;
  justify-content: space-around;
}

.space-logo {
  width: 40px;
  height: auto;
  margin-right: 10px;
  cursor: pointer;
}

ul {
  list-style-type: none;
  display: flex;
  margin-right: 2rem;
}

span {
  padding: 0 10px;
  display: block;
  cursor: pointer;
  color: var(--color-blue);
  font-size: 1rem;
}

.myprofile {
  border-left: 2px solid black;
}

.title-logo {
  font-size: 1.5rem;
  font-weight: 600;
  color: var(--color-black);
}

@media screen and (min-width: 768px) {
  header {
    height: 55px;
    justify-content: space-between;
  }

  .space-logo {
    width: 60px;
    height: auto;
    margin-right: 10px;
    cursor: pointer;
  }

  .title-logo {
    font-size: 2.25rem;
    font-weight: 600;
  }

  span {
    font-size: 1.25rem;
  }
}
