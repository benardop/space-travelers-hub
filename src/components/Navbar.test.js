import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

const MockNavbar = () => (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

test('renders correctly', () => {
  const tree = renderer
    .create(<MockNavbar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
