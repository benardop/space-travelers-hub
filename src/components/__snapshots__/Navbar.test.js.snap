// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header>
  <div
    className="logo"
  >
    <img
      alt="logo"
      className="space-logo"
      src="planet.png"
    />
    <span
      className="title-logo"
    >
      Space Travelers'  Hub
    </span>
  </div>
  <nav>
    <ul>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "bold",
            "textDecoration": "underline",
          }
        }
      >
        <span
          className="rockets"
        >
          Rockets
        </span>
      </a>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "normal",
            "textDecoration": "none",
          }
        }
      >
        <span
          className="missions"
        >
          Missions
        </span>
      </a>
      <a
        className=""
        href="/myprofile"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "normal",
            "textDecoration": "none",
          }
        }
      >
        <span
          className="myprofile"
        >
          My Profile
        </span>
      </a>
    </ul>
  </nav>
</header>
`;
